docs/scope-and-objectives.md

```diff
@@ -0,0 +1,167 @@
+- To manage the lifecycle of infrastructure unrelated to the running of Kubernetes-conformant clusters.
+- To force all Kubernetes lifecycle products (kops, kubespray, GKE, AKS, EKS, IKS etc.) to support or use these APIs.
+- To manage non-Cluster API provisioned Kubernetes-conformant clusters.
+- To manage a single cluster spanning multiple infrastructure providers.
+- To configure a machine at any time other than create or upgrade.
+- To duplicate functionality that exists or is coming to other tooling, e.g., updating kubelet configuration (c.f. dynamic kubelet configuration), or updating apiserver, controller-manager, scheduler configuration (c.f. component-config effort) after the cluster is deployed.
+
+## Requirements
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
+
+- Cluster API MUST provide isolation between providers.
+
+- Cluster API MUST support multiple infrastructure providers, including both on-prem and cloud providers.
+
+- Cluster API MUST be able to manage groups/sets of Kubernetes nodes that support scaling and orchestrated upgrades.
+
+- Cluster API, through a single installation, MUST be able to manage (create, scale, upgrade, destroy) clusters on multiple providers.
+
+- Cluster API SHOULD support all operating systems in scope for Kubernetes conformance.
+
+#### User Experience
+
+- Cluster API MUST be able to display the versions of Kubernetes and related components of a Node that it manages.
+
+- Cluster API MUST be able to provide sufficient information for a consumer to access a provisioned Kubernetes cluster’s API server.
+
+#### Organization
+
+- Cluster API MUST NOT have code specific to a particular provider within the main repository, defined as sigs.k8s.io/cluster-api.
+
+#### Validation
+
+- Cluster API provider implementations SHOULD validate configuration data.
+
+- Cluster API MUST test integrations that it claims will work.
+
+- Cluster API MUST deploy clusters that pass Kubernetes conformance tests.
+
+- Cluster API MUST document how to bootstrap itself and MAY provide tooling to do so.
+
+- Cluster API MUST document requirements for machine images and SHOULD reference default tools to prepare machine images.
+
+- Cluster API MUST have conformance tests to ensure that Cluster API and a given provider fulfill the expectations of users/automation.
+
+#### Extension
+
+- Cluster API SHOULD allow default implementations to be pluggable/extensible.
+   - Cluster API SHOULD offer default implementations for generic operations.
+   - Users MAY use different provider implementations for different sets of operations.
+
+- 🔌 Cluster API MUST be able to provide information as to which versions of Kubernetes it can install.
+
+- 🔌 Cluster API MUST be able to bootstrap a new Kubernetes control plane.
+
+- 🔌 Cluster API MUST provide a generic image lookup mechanism that can filter on characteristics such as kubernetes version, container runtime, kernel version, etc.
```

Comment:
can you give more context on where in the workflow would be this leveraged?

Reply:
This would be used where "golden images" are used, such as in the AWS provider.

The idea would be that different pre-baked images could be available for the different characteristics that are exposed and then the correct one would be used (without having to manually specify the image id) based on the requested input characteristics.

Reply:
What is going to be asking for the list of images?

Reply:
Tough to say at this point, since we haven't fully fleshed out the designs as part of the workstreams yet, but If I had to spitball, I would think it would be part of the common machine controller.

Reply:
OK. This feels a bit like it's building an image catalog service into the cluster-api. How many different variations would we expect a single cluster-api instance to need to manage? Is this meant to be a front-end of some sort that would just proxy queries to a cloud's image catalog? What if there's only ever one answer to return, like might be the case for an on-premise cluster?

Reply:
I think these are great questions and should probably be brought up as part of the `node-lifecycle-management` workstream.